pkg/deploy/api/v1beta3/types.go

```diff
@@ -254,6 +254,11 @@ type DeploymentConfigSpec struct {
 	// Replicas is the number of desired replicas.
 	Replicas int `json:"replicas" description:"the desired number of replicas"`
 
+	// Test ensures that this deployment config will have zero replicas except while a deployment is running. This allows the
+	// deployment config to be used as a continuous deployment test - triggering on images, running the deployment, and then succeeding
+	// or failing. Post strategy hooks and After actions can be used to integrate successful deployment with an action.
+	Test bool `json:"test" description:"if true, this deployment config will always be scaled to 0 except while a deployment is running"`
```

Comment:
Nope, it applies to any kind of field

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#optional-vs-required

> Using the omitempty tag causes swagger documentation to reflect that the field is optional.
Using a pointer allows distinguishing unset from the zero value for that type. There are some cases where, in principle, a pointer is not needed for an optional field since the zero value is forbidden, and thus implies unset.